### ISSUE TITLE
Lots of bugs now in mob-generation branch

### ISSUE THREAD

**Issue by K-Mistele** [amistele/ascii-adventure#25]
@Michael-N @amistele 
Help needed squishing bugs in the `mob-generation` branch. Please help! Switch to the branch, try moving into and attacking mobs. Strange bugs.

**Comment by K-Mistele** [comment]
Currently Observed Phenomenon: (most happen when lots of mobs are in close proximity to player)
- mobs can "overlap" each other, causing one to overwrite the other and make it disappear until later when both move apart
    - I suspect it has something to do with the order in which mobs and players detect collisions and try and move
- mobs can overwrite the player if they move on top of it (only happens sometimes. it shouldn't be able to happen, but I haven't determined the triggering circumstance yet. 
- if player attacks mob in some circumstances, it can appear to move diagonally, and then trying to attack it again results in an error:
`Traceback (most recent call last):`
 ` File "adventure.py", line 270, in <module>
    player_move(player_input)`
  `File "adventure.py", line 205, in player_move`
    `print("You attacked a {}!".format(detect_attack_on_mob("y", 1)[1]))`
`IndexError: list index out of range`
- now, mobs won't attack player

**Comment by K-Mistele** [comment]
@Michael-N I'm kind of at a loss where to go. Should we roll the branch back a commit?

**Comment by Michael-N** [comment]
### 1st Bullet point:
+ the mob collision detection does not handle conflicts for scenario where same position is the next move of two mobs.  The last mob to update the world tile is what is displayed... 
 
### 2nd Bullet point
+ unable to replicate

### 3rd Bullet point:
```
Traceback (most recent call last):
File "adventure.py", line 270, in <module> player_move(player_input)
File "adventure.py", line 205, in player_move
print("You attacked a {}!".format(detect_attack_on_mob("y", 1)[1]))
IndexError: list index out of range
```

+ The problem is the ' if statements'  in  function detect_attack_on_mob on ln 149 of adventure.py
+ Only one item is appended to list in some cases yet the 2nd item i.e  the item with index 1 is accessed...
+  detect_attack_on_mob  is evaluated multiple times.. where context for the came world can change between successive calls for access to the 2nd item of the list it returns... 

### 4th Bullet point
+ mobs do not attack player because detect_collision has the same errors as did  function detect_attack_on_mob
+ fix: refactor how the function is used and where it is called... also the if statements...

**Comment by Michael-N** [comment]
TL;DR; the errors are in adventure.py .... not so much the mob classes

**Comment by Michael-N** [comment]
@K-Mistele

**Comment by K-Mistele** [comment]
Player attacking mobs now works. I'm wondering if half of these problems would be partially fixed by re-drawing the board after player moves and mob moves..... I'll keep working on that.
